File: src/main/java/com/Project1/repository/UserRepository.java
package com.Project1.repository;

import com.Project1.models.User;
import java.util.Optional;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends CassandraRepository<User, String> {
    User findByEmail(String email);

    void deleteByUsername(String username);

    Optional<User> findByUsername(String userName);
}

File: src/main/kotlin/org/ProjectV1/repository/UserRepository.kt
package org.ProjectV1.repository

import org.ProjectV1.models.User
import org.springframework.data.cassandra.repository.AllowFiltering
import org.springframework.data.cassandra.repository.CassandraRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : CassandraRepository<User, String> {

    fun findByEmail(email: String): User?

    fun deleteByUsername(username: String?)

    fun findByUsername(userName: String): Optional<User>
}